src/services/development.ts

```diff
@@ -1,36 +1,102 @@
-import { mocks } from '@/api/mocks'
+import { setupWorker, RestHandler, MockedRequest } from 'msw'
+
+import { TOKENS as PROD_TOKENS, services as prodServices } from './production'
+import { merge, build, ServiceDefinition, token, get } from './utils'
+import { useBootstrap } from '../index'
+import { mocks, setupHandlers } from '@/api/mocks'
+import { fs, fakeApi } from '@/api/mocks/index'
 import CookiedEnv from '@/services/env/CookiedEnv'
-import KumaApi from '@/services/kuma-api/KumaApi'
-import { mockApi } from '@/services/kuma-api/MockKumaApi'
 import Logger from '@/services/logger/DatadogLogger'
 import { disabledLogger } from '@/services/logger/DisabledLogger'
+import type { TokenType } from '@/services/utils'
 
 export { constant, get, container, createInjections, build, merge } from './utils'
-export { TOKENS } from './production'
 
-export const services: ServiceDefinition[] = merge(prodServices, [
+type Msw = ReturnType<typeof setupWorker>
+const $ = {
+  ...PROD_TOKENS,
+  msw: token<Msw>('msw'),
+  mswKumaHandlers: token<RestHandler[]>('msw.kuma.handlers'),
+  mswKumaMocks: token<RestHandler[]>('msw.kuma.mocks'),
+  mswHandlers: token<RestHandler[]>('msw.handlers'),
+}
+
+type Env = TokenType<typeof $.Env>
+export const services: ServiceDefinition[] = [
+
   [$.Env, {
     service: CookiedEnv,
     arguments: [
       $.EnvVars,
     ],
   }],
-  [$.mocks, {
-    constant: mocks,
+
+  [$.logger, {
+    service: disabledLogger(Logger),
+    arguments: [
+      $.Env,
+    ],
   }],
-  [$.api, {
-    service: mockApi(KumaApi),
+
+  // Mock Service Worker
+  [$.msw, {
+    service: (handlers: RestHandler[]) => {
+      const worker = setupWorker(...handlers)
+      worker.start({
+        quiet: true,
+        onUnhandledRequest(req: MockedRequest) {
+          // Ignores warnings about unhandled requests.
+          if (
+            req.url.pathname.startsWith('/@fs') ||
+            req.url.pathname.startsWith('/node_modules') ||
+            req.url.pathname.startsWith('/src/assets') ||
+            req.url.href.match(/\.(vue|ts|js|json)(\?.*)?$/)
+          ) {
+            return
+          }
+
+          console.warn('Found an unhandled %s request to %s', req.method, req.url.href)
+        },
+      })
+    },
+    arguments: [
+      $.mswHandlers,
+    ],
+  }],
+
+  // old style kuma mocks
+  [$.mswKumaMocks, {
+    service: (env: Env) => setupHandlers(env.var('KUMA_API_URL'), mocks, env),
     arguments: [
       $.Env,
     ],
+    labels: [
+      $.mswHandlers,
+    ],
   }],
-  [$.logger, {
-    service: disabledLogger(Logger),
+
+  // new style kuma mocks
+  [$.mswKumaHandlers, {
+    service: (env: Env) => fakeApi(env.var('KUMA_API_URL'), fs),
     arguments: [
       $.Env,
     ],
+    labels: [
+      $.mswHandlers,
+    ],
   }],
-])
-build(services)
+]
+
+// straight-forwards bootstrap decorator
+const bootstrap: ServiceDefinition | undefined = prodServices.find(([token, _]) => token === $.bootstrap)
```

Comment:
**Suggestion**: This type is already correctly inferred.

```suggestion
const bootstrap = prodServices.find(([token, _]) => token === $.bootstrap)
```

Reply:
Done!